### ISSUE TITLE
Tasker plugin

### ISSUE THREAD

**Issue by jumper047** [SamAmco/track-and-graph#6]
It is useful to have ability to mark habits automatically. For example, in my current habits tracker  i automatically mark as done habit "Dont smoke today").

**Comment by SamAmco** [comment]
Hi jumper047. Sorry I'm not sure I fully understand the use case here. You could try using timestamp only if you just want to quickly mark something in time?

**Comment by jumper047** [comment]
Ohh, sorry, i'll try to explain better. Case 1: i track event, which is very rare at moment when i start track it, but became almost regular at this moment. So with tasker integration i can set mark "event occurs" every day automatically, and manually delete point if needed.
Case 2: With tasker integration i can set event marks from my watch (pebble - it has some tasker apps).

**Comment by SamAmco** [comment]
Sorry I have closed this issue due to a lack of interest and clarity.

**Comment by Wilker-a** [comment]
basically a point in time represented by a line somewhere in the horizontal axis (or whatever one that represents time), which can be marked by another application. is that right?

**Comment by jumper047** [comment]
Yes - from user's perspective it is something like "set point with current time when i press button on my watch". Watch's button is just example, trigger can be any event in Tasker: time, location, whatever.

**Comment by SamAmco** [comment]
I believe this functionality is accounted for by default values and/or plot totals. So for the example of smoking you could use a numeric input that represents the number of cigarettes you smoked that day. Then only add inputs if and when you do smoke. In order to graph this you would create a line graph with "Plot daily totals" set such that it will plot a point for every day representing the sum of all cigarettes smoked that day. If there was no input for a day then the sum is zero. 

Alternatively if you really want those zeros in there you could set add a default value of zero to your numeric input. Then you can tap the plus button and automatically add a zero. If you want to change it you can go in and do so manually. (default values haven't yet been released but I expect they will some time this week.)

**Comment by jumper047** [comment]
Ohh, sorry, I confused you with cigarettes example - it is actually corner case. I mentioned it just because it reveals power of tasker, it can make life easier in many unexpected situations. 
I think more interesting possibility here - integration with weareble devices via tasker. If you count some medical sympthoms, or, for example, yellow cars you encountered during day - It is much convenient to add mark about some event from smartwatch/smartband without need to pick up your smartphone from pocket. There are a lot of wearables, from cheap smartbands to smartwatches with Android wear who can be integrated with tasker. As far as i know, all xiaomi devices have tasker support via unofficial apps - so you can add points to Track&Graph via cheap miband 4 for example. As of my experience it is whery usefull

**Comment by SamAmco** [comment]
Oh I see! I'm very sorry I was not aware of what Tasker was. I believe I understand what you're talking about now. I will re-open this issue but I think this might be one for someone else to take on as I don't have a smartwatch and I have other priorities in the app right now.

**Comment by dreamflasher** [comment]
@SamAmco Thank you for your great app and thank you for reopening the ticket. I came here because I want to track habits (noticing) with a hardware button of my phone; Tasker/AutoInput can recognize that, but now I would need some way to externally (could be an Intent) tell your app "track this activity". If not a full tasker plugin, would be great if you at least could an intent?

**Comment by SamAmco** [comment]
Hi @dreamflasher unfortunately I really have very little time for this project at the moment. However if you know what an intent is, perhaps you know enough about android to have a go yourself?

**Comment by aleksfadini** [comment]
I second this request. With tasker, it would be possible to track a habit while completing another action (example: set a pomodoro timer, when it expires it would automatically tick a track and graph habit). 

We need either a tasker plugin, or the shortcut when the app is long pressed to allow to directly track a habit. For now we have a widget on long press (which is great). 

I can open another request for direct tracking of a single data on long press if it helps.
